src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/BaseRepository.cs

```diff
@@ -82,6 +83,7 @@ private async Task<DocumentClient> CreateDatabaseAndCollection(CancellationToken
             {
                 ConnectionMode = ConnectionMode.Gateway,
                 ConnectionProtocol = Protocol.Https,
+                IdleTcpConnectionTimeout = new TimeSpan(0, 10, 0)
```

Comment:
Is this the correct setting to use? I thought RequestTimeout was the one affecting this to fail as the tests gets a timeout after 60 seconds. This setting will free up tcp connections that have been idle for more than 10 minutes and I guess the test is running.

Reply:
There is an underlying socket exception towards cosmos. Based on what I have found online, this has been helpful for others that have experience the same error, so I would like to see if it has any effect for us. [Here's one of the threads discussing it. ](https://stackoverflow.com/questions/57604703/azure-cosmos-db-throwing-socket-exceptions)

Reply:
I'm a bit curious on why we end up with idle tcp connections in the first place, sounds like the client isn't closing them if we end up with filling up the connection pool. Shouldn't be any harm in closing them after 10 minutes, although there might be firewalls sitting in front that actually does this already.